lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1213,14 +1213,16 @@ def _create_node_runbook(
                 azure_node_runbook.location, azure_node_runbook.marketplace
             )
             # HyperVGenerationTypes return "V1"/"V2", so we need to strip "V"
-            azure_node_runbook.hyperv_generation = _get_vhd_generation(image_info)
+            if image_info:
+                azure_node_runbook.hyperv_generation = _get_vhd_generation(image_info)
 
             # retrieve the os type for arm template.
-            if azure_node_runbook.is_linux is None:
-                if image_info.os_disk_image.operating_system == "Windows":
-                    azure_node_runbook.is_linux = False
-                else:
-                    azure_node_runbook.is_linux = True
+            if (
+                azure_node_runbook.is_linux is None
+                and image_info
+                and image_info.os_disk_image.operating_system == "Windows"
+            ):
+                azure_node_runbook.is_linux = False
```

Comment:
Why remove the `azure_node_runbook.is_linux = False`?

Reply:
I removed the lines
``` Python
else:
    azure_node_runbook.is_linux = True
```
because Flake8 was complaining the logic was too complicated and the default value is already True.
See later in the function:
``` Python
if azure_node_runbook.is_linux is None:
    # fill it default value
    azure_node_runbook.is_linux = True
```